src/indexer/segment_updater.rs

```diff
@@ -46,6 +47,7 @@ const NUM_MERGE_THREADS: usize = 4;
 pub fn save_new_metas(schema: Schema, directory: &dyn Directory) -> crate::Result<()> {
     save_metas(
         &IndexMeta {
+            index_settings: IndexSettings::default(),
```

Comment:
@fulmicoton 
Can we load the index metas from Index to access index_settings or does it not yet exist?

If it doesn't exist, we would need to either write them upfront or move IndexMeta into Index to have a place to store index settings on creation.

Reply:
I would give it the same status as schema.

Load it from IndexMeta and store it as a member of Index.